hw/ip/usbuart/data/usbuart.hjson

```diff
@@ -2,51 +2,127 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 { name: "usbuart",
-  clock_primary: "clk_fixed",
-  other_clock_list: [ "clk_48mhz" ]
+  clock_primary: "clk_i",
+  other_clock_list: [ "clk_usb_48mhz_i" ]
+  reset_primary: "rst_ni",
+  other_reset_list: [ "rst_usb_48mhz_ni" ]
   bus_device: "tlul",
   bus_host: "none",
   available_inout_list: [
-    { name: "usb_dp", desc: "USB data D+" }
-    { name: "usb_dm", desc: "USB data D-" }
+    { name: "d", desc: "USB data differential" }
+    { name: "dp", desc: "USB data D+" }
+    { name: "dn", desc: "USB data D-" }
   ],
   available_input_list: [
-    {  name: "usb_sense", desc: "USB host VBUS sense" }
+    {  name: "sense", desc: "USB host VBUS sense" }
   ],
   available_output_list: [
-    { name: "usb_pullup", desc: "USB FS pullup control" }
+    { name: "se0", desc: "USB single-ended zero link state" }
+    { name: "dp_pullup", desc: "USB D+ pullup control" }
+    { name: "dn_pullup", desc: "USB D- pullup control" }
+    { name: "tx_mode_se", desc: "USB single-ended transmit mode control" }
+    { name: "suspend", desc: "USB link suspend state" }
+  ],
+  inter_signal_list: [
+    { name:    "usb_ref_val",
+      type:    "uni",
+      act:     "req",
+      package: "",
+      struct:  "logic",
+      width:   "1"
+    }
+    { name:    "usb_ref_pulse",
+      type:    "uni",
+      act:     "req",
+      package: "",
+      struct:  "logic",
+      width:   "1"
+    }
+  ]
+  param_list: [
+    { name:    "NEndpoints",
+      type:    "int",
+      default: "12",
+      desc:    "Number of endpoints",
+      local:   "true"
+    }
   ],
   interrupt_list: [
```

Comment:
I bounced back and forth on this one.

Given the usbuart is not in the official earlgrey it is going to be lower priority, so I opted for what seemed the simplest way of being able to switch it in to do quick tests that it is working. (I did actually wonder about doing a new top that just has the I/O blocks) and with my approach (keeping both in top_earlgray.core) switching back and forth (which I was doing) only needs a single edit in the autogen top_earlgrey.sv

I do want to evaluate all the interrupts a bit more carefully and see if it would be an advantage to reorder the ones in usbdev. Potentially the ones to do with buffers and transmit/receive (with no equivalent in usbuart) could line up with the uart ones. And the ones about the USB interface (errors etc) could be common to usbdev/usbuart. A bit like the extra registers, if you direct UART code at the interface it makes no use of the extra interrupts , but code that is aware it is using usbuart could use them. But I do want to make sure (and eventually test more carefully) that plain UART code would work.